### PR TITLE
[IIS] Move to GenericHost

### PR DESCRIPTION
**Pull request by Kahbazi**
Part of #20964

### CODE REVIEW

**Comment by jkotalik** [comment]
Looks like a lot of tests are failing in the IISIntegration middleware tests, for example MiddlewareShutsdownGivenANCMShutdown. Can you look into why those are failing?

**Comment by Tratcher** [comment]
Test failures:
https://dev.azure.com/dnceng/public/_build/results?buildId=744553&view=ms.vss-test-web.build-test-results-tab&runId=23076178&resultId=102339&paneView=debug

It's missing a call to StartAsync()

**Comment by Tratcher** [comment]
A few test failures.
https://dev.azure.com/dnceng/public/_build/results?buildId=744859&view=ms.vss-test-web.build-test-results-tab&runId=23085538&paneView=debug

**Comment by Kahbazi** [comment]
I'm having a difficult time fixing the tests, I think the reason is the different implementation of `Dispose` method between `Host` and `WebHost`. 
When `WebHost` is being disposed, it will also call `StopAsync()` which also stops the `IServer`,
but when Host is being disposed, it won't call `StopAsync()`. 
Is this behavior by design? 
Am I on the right path?

File: src/Servers/IIS/IIS/test/IIS.Tests/Utilities/TestServer.cs [hunk 1]
```diff
@@ -131,15 +131,20 @@ private void InitializeConfig(int port)
...
-            var builder = new WebHostBuilder()
-                .UseIIS()
+            var builder = new HostBuilder()
+                .ConfigureWebHost(webHostBuilder =>
+                {
+                    webHostBuilder
+                        .UseIIS()
+                        .UseSetting(WebHostDefaults.ApplicationKey, typeof(TestServer).GetTypeInfo().Assembly.FullName);
+                })
                 .ConfigureServices(services =>
```
Comment (BrennanConroy) [inline src/Servers/IIS/IIS/test/IIS.Tests/Utilities/TestServer.cs:141]: This might need to move to the `webHostBuilder`

**Review by BrennanConroy — APPROVED** [review]

**Comment by BrennanConroy** [comment]
Thanks!

**Pull request merged**